lighthouse-core/runner.js

```diff
@@ -11,11 +11,14 @@ const ReportGeneratorV2 = require('./report/v2/report-generator');
 const Audit = require('./audits/audit');
 const emulation = require('./lib/emulation');
 const log = require('lighthouse-logger');
+const assetSaver = require('./lib/asset-saver');
 const fs = require('fs');
 const path = require('path');
 const URL = require('./lib/url-shim');
 const Sentry = require('./lib/sentry');
 
+const basePath = path.join(process.cwd(), 'latest-run');
+
 class Runner {
   static run(connection, opts) {
```

Comment:
this method is a bit big for its 👖  these days, it seems like it can roughly be broken down into

* loadOrGatherArtifacts
* potentiallySaveArtifacts
* potentiallyComputeAudits
* combine results

do you think it'd be easier to handle the `config.gatherMode` type flags individually in each of these 4 hypothetical methods rather than exploding the matrix of choices? the plain english variable names of "this is what is happening" is kinda nice right now, but its quite a long function

Reply:
Agreed on no real need to make a variable for each possible state. Rather than just four, can't we just have
- loadArtifacts
- gatherArtifacts
- saveArtifacts
- runAudits
- score
- createLHR
and just have run() run the ones that are needed for each flag?

Reply:
sg. say hello to

_loadArtifactsFromDisk
_gatherArtifactsFromBrowser
_saveArtifacts
_runAudits
_scoreAndCategorize